src/integration/java/io/pinecone/integration/dataPlane/UpsertDescribeIndexStatsAndDeletePodTest.java

```diff
@@ -6,124 +6,106 @@
 import static io.pinecone.helpers.BuildUpsertRequest.*;
-import static io.pinecone.helpers.IndexManager.createIndexIfNotExistsDataPlane;
 import static org.junit.jupiter.api.Assertions.assertEquals;
 
 public class UpsertDescribeIndexStatsAndDeletePodTest {
+    private static final TestIndexResourcesManager indexManager = TestIndexResourcesManager.getInstance();
+    private static Pinecone pineconeClient = new Pinecone.Builder(System.getenv("PINECONE_API_KEY")).build();
     private static Index index;
     private static AsyncIndex asyncIndex;
-    private static final int dimension = 3;
+    private static String indexName;
+    private static int dimension;
     private static final Struct nullFilterStruct = null;
+    private static final String namespace = RandomStringBuilder.build("ns", 8);
+    private static int namespaceVectorCount = 0;
 
     @BeforeAll
     public static void setUp() throws IOException, InterruptedException {
-        AbstractMap.SimpleEntry<String, Pinecone> indexAndClient = createIndexIfNotExistsDataPlane(dimension, IndexModelSpec.SERIALIZED_NAME_POD);
-        String indexName = indexAndClient.getKey();
-        Pinecone pineconeClient = indexAndClient.getValue();
+        indexName = indexManager.getPodIndexName();
+        dimension = indexManager.getDimension();
         index = pineconeClient.getIndexConnection(indexName);
         asyncIndex = pineconeClient.getAsyncIndexConnection(indexName);
     }
 
     @Test
     public void upsertVectorsAndDeleteByIdSyncTest() throws InterruptedException {
         // Upsert vectors with required parameters
-        int dimension = 3;
-        Struct emptyFilterStruct = null;
         int numOfVectors = 3;
-
-        String namespace = RandomStringBuilder.build("ns", 8);
         List<String> upsertIds = getIdsList(numOfVectors);
+        List<VectorWithUnsignedIndices> vectorsToUpsert = new ArrayList<>(numOfVectors);
+
         for (String id : upsertIds) {
-            index.upsert(id,
-                    generateVectorValuesByDimension(dimension),
-                    namespace);
+            VectorWithUnsignedIndices vector =
+            new VectorWithUnsignedIndices(id, generateVectorValuesByDimension(dimension));
+            vectorsToUpsert.add(vector);
         }
 
-        int actualVectorCount = numOfVectors;
-
-        // wait sometime for the vectors to be upserted
-        Thread.sleep(10000);
+        index.upsert(vectorsToUpsert, namespace);
+        namespaceVectorCount += numOfVectors;
 
         assertWithRetry(() -> {
             // call describeIndexStats to get updated vector count
             DescribeIndexStatsResponse describeIndexStatsResponse = index.describeIndexStats();
-
+            System.out.println("ASSERT WITH RETRY 1");
             // verify the updated vector count
-            assertEquals(describeIndexStatsResponse.getNamespacesMap().get(namespace).getVectorCount(), actualVectorCount);
-        });
-
-        // Delete 1 vector
```

Comment:
I'm not sure why we were deleting one vector, then waiting, then deleting the others. It was confusing to read and work around. I batched things into one `index.delete()` request down below. Faster and easier to reason about, and it still tests the same code paths.